Build native image inside docker fails when using JDK8

### Expected Behavior

It should be possible to build a native image inside docker even if using Java 8 locally.

### Actual Behaviour

When executing `./gradlew dockerBuildNative` from the command line when using Java 8, the generation of the native image inside docker fails because there is no GraalVM for Java 8 anymore:

```
> Task :dockerBuildNative FAILED
Building image using context '/home/ivan/workspaces/micronaut-graal-tests/micronaut-basic-app/build/docker'.
Using Dockerfile '/home/ivan/workspaces/micronaut-graal-tests/micronaut-basic-app/build/docker/DockerfileNative'
Using images 'basic-app'.
Step 1/11 : FROM ghcr.io/graalvm/graalvm-ce:java8-21.3.0 AS graalvm

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dockerBuildNative'.
> Could not build image: manifest unknown
```

We need to upgrade to GraalVM JDK11/17, also we should apply this change from master in 2.0.x branch: https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/274/files#diff-c55bcfa6c1152b327585901498c937fa0d5aa5ac9b13a8d7237002eb6e68ef51R312


### Steps To Reproduce

- `git clone https://github.com/micronaut-graal-tests/micronaut-basic-app`
- `cd micronaut-basic-app`
- `git checkout 3.2.x`
- Make sure you use Java 8
- `./gradlew dockerBuildNative`

### Environment Information

Linux Mint 20.2 & JDK 8

### Example Application

_No response_

### Version

3.2.0-SNAPSHOT and Micronaut Gradle plugin 2.0.7

--- a/src/main/java/io/micronaut/gradle/docker/NativeImageDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/NativeImageDockerfile.java
@@ -52,7 +52,7 @@ public abstract class NativeImageDockerfile extends Dockerfile implements Docker
 
     private static final List<Integer> SUPPORTED_JAVA_VERSIONS = Collections.unmodifiableList(
             // keep those in descending order
-            Arrays.asList(16, 11, 8)
+            Arrays.asList(17, 11)
     );
 
     /**

